File: mobile/ios/NavTin/Multiplayer/SupabaseService.swift
import Foundation

import Foundation

class SupabaseService {
    static let shared = SupabaseService()
    
    // Values will be read from Secrets.xcconfig in a real production build
    private let url = "https://ofplxtwqnpechpwodqor.supabase.co"
    private let key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im9mcGx4dHdxbnBlY2hwd29kcW9yIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Njk0MjY2NjUsImV4cCI6MjA4NTAwMjY2NX0.RdBpNIOtgwhbf_ZYFPOqDb0DRFXhqyjvuq3llAZ3Omw"
    
    func startMatchmaking(userId: String, name: String, completion: @escaping (String, PlayerSide) -> Void) {
        print("Matchmaking started for \(name)...")
        
        // 1. Join 'matchmaking' channel
        // 2. Insert into 'matchmaking_queue'
        // 3. Listen for matched room_id
        
        // Simplified flow for MVP:
        // Try to find an open room or create one
        Task {
            // This is a placeholder for the actual PostgREST / Realtime calls
            // Implementation requires 'supabase-swift' package
            print("Supabase: Matchmaking logic executing...")
            
            // Mocking the result of a successful join
            DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
                completion("room_active_swift", .white)
            }
        }
    }
    
    func sendMove(roomId: String, state: GameState) {
        print("Supabase: Sending move to room \(roomId)")
        // Implementation: Update 'game_rooms' table with new state
        // supabase.from("game_rooms").update(values: ["game_state": state.toJSON()]).eq("id", roomId)
    }
    
    func listenForOpponentMoves(roomId: String, onMove: @escaping () -> Void) {
        print("Supabase: Subscribing to room \(roomId)")
        // Implementation: Realtime subscription to 'game_rooms' for 'UPDATE' events
    }
    
    func deleteRoomIfEmpty(roomId: String) {
        print("Supabase: Checking room occupancy for cleanup")
        // Implementation: Rpc call to 'delete_game_room_data'
    }
}

File: mobile/android/app/src/main/java/com/navtin/multiplayer/SupabaseService.kt
package com.navtin.multiplayer

import com.navtin.engine.GameState
import com.navtin.engine.PlayerSide
import kotlinx.coroutines.delay

object SupabaseService {
    private const val URL = "https://ofplxtwqnpechpwodqor.supabase.co"
    private const val KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im9mcGx4dHdxbnBlY2hwd29kcW9yIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Njk0MjY2NjUsImV4cCI6MjA4NTAwMjY2NX0.RdBpNIOtgwhbf_ZYFPOqDb0DRFXhqyjvuq3llAZ3Omw"

    suspend fun startMatchmaking(userId: String, name: String, onMatched: (String, PlayerSide) -> Unit) {
        println("Supabase: Matchmaking transition for $name")
        
        // Implementation using supabase-kt Postgrest & Realtime:
        // val client = createSupabaseClient(URL, KEY) { ... }
        // client.postgrest["matchmaking_queue"].insert(...)
        
        delay(2000)
        onMatched("room_android_real", PlayerSide.BLACK)
    }

    fun sendMove(roomId: String, state: GameState) {
        println("Supabase: Sending move to $roomId")
        // client.postgrest["game_rooms"].update(...)
    }

    fun subscribeToRoom(roomId: String, onUpdate: () -> Unit) {
        println("Supabase: Subscribing to real-time updates for $roomId")
        // val channel = client.realtime.createChannel("room-$roomId")
        // channel.broadcast(...)
    }
    
    fun cleanupRoom(roomId: String) {
        println("Supabase: Calling room cleanup RPC")
        // client.postgrest.rpc("delete_game_room_data", ...)
    }
}
